addons/hw_drivers/iot_handlers/interfaces/DisplayInterface_L.py

```diff
@@ -22,28 +26,41 @@ def get_devices(self):
 
         if screeninfo is None:
             # On IoT image < 24.08 we don't have screeninfo installed, so we can't get the connected displays
-            # We return a single display with x_screen = 0, to open a browser anyway, in case one is connected
-            display_identifier = 'hdmi_0'
-            display_devices[display_identifier] = {
-                'identifier': display_identifier,
-                'name': 'Display - ' + display_identifier,
-                'x_screen': '0',
-            }
-            return display_devices
+            # We use old method to get the connected display
+            hdmi_ports = {'hdmi_0': 2, 'hdmi_1': 7} if 'Pi 4' in GPIO.RPI_INFO.get('TYPE') else {'hdmi_0': 2}
+            try:
+                for x_screen, port in enumerate(hdmi_ports):
+                    if Vcgencmd().display_power_state(hdmi_ports.get(port)) == 'on':
+                        display_devices[port] = self._add_device(port, x_screen)
+                return display_devices
+            except subprocess.CalledProcessError:
+                _logger.warning('Vcgencmd "display_power_state" method call failed')
 
         try:
             os.environ['DISPLAY'] = ':0'
             for x_screen, monitor in enumerate(screeninfo.get_monitors()):
-                display_identifier = monitor.name
-                display_devices[display_identifier] = {
-                    'identifier': display_identifier,
-                    'name': 'Display - ' + display_identifier,
-                    'x_screen': str(x_screen),
-                }
+                display_devices[monitor.name] = self._add_device(monitor.name, x_screen)
+            return display_devices
         except screeninfo.common.ScreenInfoError:
             # If no display is connected, screeninfo raises an error, we return the distant display
-            display_devices['distant_display'] = {
-                'name': "Distant Display",
-            }
+            display_devices['distant_display'] = self._add_device('distant_display', 0, distant=True)
```

Comment:
Can be simplified with a `dummy_display` variable
```suggestion
            return dummy_display
```